src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -60,10 +78,15 @@ object Scope {
    * @throws UnsupportedOperationException if the ast is not a workflow ast
    * @return a workflow
    */
-  def generateWorkflow(ast: Ast, namespaces: Seq[WdlNamespace], tasks: Seq[Task], wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Workflow = {
+  def generateWorkflow(ast: Ast,
+                       namespaces: Seq[WdlNamespace],
+                       tasks: Seq[Task],
+                       wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter): Workflow = {
     ast.getName match {
-      case AstNodeName.Workflow => generateTree(ast, namespaces, tasks, wdlSyntaxErrorFormatter).asInstanceOf[Workflow]
-      case nonWorkflowAst => throw new UnsupportedOperationException(s"Ast is not a 'Workflow Ast' but a '$nonWorkflowAst Ast'")
+      case AstNodeName.Workflow =>
+        generateWorkflowScopes(ast, namespaces, tasks, wdlSyntaxErrorFormatter)
```

Comment:
ok there is a check for workflowedness here, but not in the `generateWorkflowScopes` method